gson/src/main/java/com/google/gson/GsonBuilder.java

```diff
@@ -696,8 +696,8 @@ public GsonBuilder registerTypeAdapter(Type type, Object typeAdapter) {
 
   private boolean isTypeObjectOrJsonElement(Type type) {
     return type instanceof Class &&
-            (type == Object.class
-                    || JsonElement.class.isAssignableFrom((Class<?>) type));
+        (type == Object.class
```

Comment:
Not very important, but if you wanted to follow Google Java Style here, the `&&` should be at the start of this line.